### DISCUSSION TITLE
Bulk creation of records

### DISCUSSION THREAD

**Discussion by Lishenga** [mikro-orm/mikro-orm#1580] [Q&A] [answered]
Could anyone assist me with code in the bulk creation of records to the database.
For instance you can have an array of records that have been sent from a frontend application and you want to populate them one by one in a foreach loop or in any efficient way. Anyone who has ideas to achieve this in an efficient way using mikroorm?

**Comment by B4nan** [comment, marked as answer]
That works automatically, just persist many entities and flush once

```ts
for (let i = 1; i <= 5; i++) {
  const u = new User(`Peter ${i}`, `peter+${i}@foo.bar`);
  em.persist(u);
}

await em.flush();

// insert into `user` (`name`, `email`) values
//   ('Peter 1', 'peter+1@foo.bar'),
//   ('Peter 2', 'peter+2@foo.bar'),
//   ('Peter 3', 'peter+3@foo.bar'),
//   ('Peter 4', 'peter+4@foo.bar'),
//   ('Peter 5', 'peter+5@foo.bar');
```
https://mikro-orm.io/blog/mikro-orm-4-1-released#batch-inserts-updates-and-deletes

**Reply by danielgak** [reply]
In your example @B4nan you are inserting new entities, but for me its still not shure what to do in case I want to insert depending on the result of a query. That is, imagine that there exists antother entity **Notification** so those users who's name is _Tom_ will recive a notification.

So with postgres the query that bulk-insert into notifications would look something like:

```sql
insert into notifications (user_id, content) 
   (select id as user_id, `hey tom!` as content from users where name = `tom`)
```

but when I tried to actually implement it with mikro-om, my aproach was to use queryBuilders as I saw some update examples in [the docs](https://mikro-orm.io/docs/query-builder#using-sub-queries) but couldnt get any result.

```typescript
const knex = orm.em.getKnex();

const toms = orm.em
      .createQueryBuilder(User)
      .select('id')
      .where({ name: 'tom' })
      .getKnex();
      
await orm.em
      .createQueryBuilder(Notification)
      .withSubQuery(toms, 'toms')
      .insert({
        user: knex.ref('toms.id'),
        content: 'hey tom!',
      })
      .execute();

//  missing FROM-clause entry for table users
```

Not sure what im doing wrong... What would be the recomended aproach in this case?
